app/controllers/ballpark_logs_controller.rb

```diff
@@ -54,4 +55,12 @@ def ballpark_log_params
     def update_ballpark_log_params
       params.require(:ballpark_log).permit(:title, :description, :image, :stadium_id, :game_id, :result)
     end
+
+    def ensure_correct_user
+      @ballpark_log = BallparkLog.find(params[:id])
+      if @ballpark_log.user_id != current_user.id
+        flash[:notice] = "権限がありません"
+        redirect_to("/")
```

Comment:
```suggestion
redirect_to root_path, alert: "権限がありません"
```

権限がない操作を行われるのはお知らせより警告のニュアンスのあるalertでメッセージを設定する方が良いかもですね！

Reply:
ちなみに権限管理は有名どころのGemがあります！

https://github.com/varvet/pundit

規模がある程度おおきくなると権限管理が複雑になってくるのでgemでシンプルに権限管理を行うということもできます！